lib/handlebars/compiler/compiler.js

```diff
@@ -156,11 +156,13 @@ Compiler.prototype = {
 
   DecoratorBlock(decorator) {
     let program = decorator.program && this.compileProgram(decorator.program);
-    let params = this.setupFullMustacheParams(decorator, program, undefined),
+    let setup = this.setupFullMustacheParams(decorator, program, undefined),
+        params = setup.params,
+        splatMap = setup.splatMap,
```

Comment:
nit: Should we use destructuring here? (I forget what exact rules Handlebars has enabled, it was one of the earlier ES6 projects for me so it is likely a bit more conservative than need be)